CAP Java 5.x Migration Guide: removed repackaged `Olingo` (#2528)
cap-js/docs: java/migration.md

  - `com.sap.cds.services.ServiceExceptionUtils.getMessageTarget(String parameter, Class<E> type, Function<E, Object> path)` → `MessageTarget.create(parameter, type, path)`
  - `com.sap.cds.services.ServiceExceptionUtils.getMessageTarget(String parameter, Function<StructuredType<?>, Object> path)` → `MessageTarget.create(parameter, path)`

### Removed repackaged Olingo Dependencies { #removed-olingo-4-to-5 }

> The internal `mvn` modules `repackaged/odata-v4-lib` and `repackaged/odata-v2-lib` are removed and not released anymore. If the project directly referenced those modules and doesn't compile after migrating to CAP Java 5.x, there are 3 options to keep the compatability of the code base.

Reviewer: ```suggestion
The internally used maven modules `repackaged/odata-v4-lib` and `repackaged/odata-v2-lib` are removed from the delivery. If the project directly references these modules and doesn't compile after migrating to CAP Java 5.x, there are 3 options to keep the compatibility of the code base:
```

1. [**Recommended**] Remove `mvn` dependencies to `repackaged/odata-v4-lib`, `repackaged/odata-v2-lib` and  rewrite code using CAP Java native APIs

2. Self-repackage and reuse the Olingo modules with `maven-dependency-plugin`

> 3. Use upstream open-source Apache Olingo. Change dependencies from corresponding internal CAP `mvn` modules to OSS `org.apache.olingo:olingo-odata4`, `org.apache.olingo:olingo-odata2`

Reviewer: the second sentence is not clear - it sounds like patching CAP libraries cds-adapter-odatav4 etc.

Author: Described a bit more in details. We can even add the xml snippet from the our ex-library, or would it be too much?

Reviewer: ah ok, got it
